test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -848,6 +995,44 @@ public void CreateValidationDetails_ValidatableObject_AlreadyInContext_Ignores()
             Assert.Same(attribute, validatorMetadata);
         }
 
+        private IEnumerable<KeyValuePair<EnumGroupAndName, string>> GetLocalizedEnumGrouopedDisplayNamesAndValues()
+        {
+            var provider = CreateIStringLocalizerProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            return context.DisplayMetadata.EnumGroupedDisplayNamesAndValues;
+        }
+
+        private DataAnnotationsMetadataProvider CreateIStringLocalizerProvider()
+        {
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(loc => loc[It.IsAny<string>()])
+                .Returns<string>((k =>
+                {
+                    if (k.Contains("Loc_Two") || k.Contains("Type_Three"))
```

Comment:
We need it to return the unaltered string (like the localizer would) for the "One" enum in order to ensure that it returns the unaltered Display string. We need it to "localize" for "Type_Three" so that we can ensure we don't actually try to use `IStringLocalizer` on a ResourceType Enum. If we unconditionally alter the string we would be unable to test the case that an enum has a `DisplayAttribute` but doesn't have an entry in the `IStringLocalizer`.